### DISCUSSION TITLE
main need to change for typescript/javascript

### DISCUSSION THREAD

**Discussion by trixobird** [vercel/turborepo#4256] [Help] [answered]
I am trying to create a proof of concept of turbo repo to promote it in my company. I have ended up where I want and here is the repository: https://github.com/trixobird/turborepo

It is just an express server that is using apart from the things in tutorial (eslint, tsconfig) a logger.

There is one thing that I cannot do. As the repo stands Dockerfile works and I have a working express server that is under 200MB, very cool. But the `pnpm --filter "express" dev` fails which makes sense since in the `packages/logger/package.json` the main points to dist/index.js. Once I change this to `./src/index.ts` the `pnpm --filter "express" dev` works, but now the Dockerfile fails (again this looks normal).

I think my problem should have an easy solution, but I am failing to find it, could you help me please?

```
apps/express/package.json
{
...
  "dependencies": {
    "express": "^4.18.2",
    "logger": "workspace:*"
  },
...
}
```

```
packages/logger/package.json
{
...
  "main": "./dist/index.js",
  "types": "./src/index.ts",
...
}
```

**Comment by SherleyDuffeeX5E** [comment]
The issue you're encountering is because of the discrepancy between the main and types fields in the packages/logger/package.json file. When you set the main field to "./dist/index.js", you're telling Node.js to use the compiled JavaScript file when importing the package. However, when you set the types field to "./src/index.ts", you're telling TypeScript to use the TypeScript file when importing the package.

**Reply by trixobird** [reply]
Yeap that the reason, my question is how to overcome it. I have two different use cases:
a) `apps/express/Dockerfile `which needs the `"main": "./dist/index.js",` and
b) `pnpm --filter express dev` which needs the `"main": "./src/index.ts",`
Is there a way to have both of them working?

**Reply by anthonyshew** [reply, marked as answer]
No, because the package needs to have its exports explicitly marked when working within the bounds of the repository.

I'd encourage you to use `./dist/index.js` in this case, and use `tsc --watch` for your dev script. That way, both builds and local development will be able to use `./dist/index.js`.
